### ISSUE TITLE
mock-fs 4.3.0 broken on node 8

### ISSUE THREAD

**Issue by AGrzes** [tschaub/mock-fs#208]
I'm using mock-fs in my tests and they fail on node 8. 
I ran the module test suite and there is a lot of errors.
Attached the test log [mock-fs@node8-test.txt](https://github.com/tschaub/mock-fs/files/1049892/mock-fs.node8-test.txt)

**Comment by AGrzes** [comment]
It looks like node 8 is throwing away the values provided to callback by mock-fs.

```
// Special case of `makeCallback()` that is specific to async `*stat()` calls as
// an optimization, since the data passed back to the callback needs to be
// transformed anyway.
function makeStatsCallback(cb) {
  if (cb === undefined) {
    return rethrow();
  }

  if (typeof cb !== 'function') {
    throw new TypeError('"callback" argument must be a function');
  }

  return function(err) {
    if (err) return cb(err);
    cb(err, statsFromValues());
  };
}
```
https://github.com/nodejs/node/blob/v8.0.0/lib/fs.js#L153

**Comment by AGrzes** [comment]
Related issue #197 and PR https://github.com/nodejs/node/pull/11665

**Comment by AGrzes** [comment]
Started work in #209

**Comment by tschaub** [comment]
Fix published in `mock-fs@4.4`.  Thanks @AGrzes.
